plugins/pagerduty/src/api/types.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Incident, OnCall, Service } from '../components/types';
+
+export interface PagerDutyClient {
+  /**
+   * Fetches a list of services, filtered by the provided integration key.
+   *
+   */
+  getServiceByIntegrationKey(integrationKey: string): Promise<Service[]>;
+
+  /**
+   * Fetches a list of incidents a provided service has.
+   *
+   */
+  getIncidentsByServiceId(serviceId: string): Promise<Incident[]>;
+
+  /**
+   * Fetches the list of users in an escalation policy.
+   *
+   */
+  getOnCallByPolicyId(policyId: string): Promise<OnCall[]>;
+
+  /**
+   * Triggers an incident to whoever is on-call.
+   */
+  triggerAlarm(
```

Comment:
For functions that take a fairly large number of arguments, it's cleaner to make it take ONE argument that is am object with these args as keys instead. Essentially `triggerAlarm(request: { integrationKey: string;...`